### PR TITLE
Feature/settings

### PR DESCRIPTION
**Pull request by MaxAstin**

### CODE REVIEW

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/login/LoginViewModel.kt [hunk 1]
```diff
@@ -79,9 +82,10 @@ class LoginViewModel @Inject constructor(
         }
     }
 
-    private fun subscribeOnToken() {
-        dataStoreRepo.token.onEach { token ->
-            if (token.isEmpty()) {
+    private fun checkToken() {
+        viewModelScope.launch {
```
Comment (xidealo) [inline app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/login/LoginViewModel.kt:86]: handler?

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/settings/SettingsFragment.kt [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        ) {
+            when (val state = uiState.state) {
+                SettingsUiState.State.Loading -> {
+                    LoadingScreen()
+                }
+                SettingsUiState.State.Error -> {
+                    ErrorScreen(
+                        mainTextId = R.string.title_common_can_not_load_data,
+                        extraTextId = R.string.msg_common_check_connection_and_retry,
+                        onClick = {}
```
Comment (xidealo) [inline app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/settings/SettingsFragment.kt:90]: empty onClick?

File: data/src/main/java/com/bunbeauty/data/repository/DataStoreRepository.kt [hunk 3]
```diff
@@ -38,62 +31,58 @@ class DataStoreRepository @Inject constructor(@ApplicationContext private val co
...
     }
 
-    override suspend fun saveCafeUuid(cafeUuid: String) {
-        withContext(IO) {
-            context.cafeUuidDataStore.edit {
-                it[CAFE_UUID_KEY] = cafeUuid
-            }
+    override suspend fun saveCompanyUuid(companyUuid: String) {
+        context.userDataStore.edit {
+            it[COMPANY_UUID_KEY] = companyUuid
```
Comment (xidealo) [inline data/src/main/java/com/bunbeauty/data/repository/DataStoreRepository.kt:55]: COMPANY_UUID_KEY == userDataStore?

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/login/LoginViewModel.kt (hunk 1 above)
Comment (MaxAstin) [reply app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/login/LoginViewModel.kt]: fixed

File: data/src/main/java/com/bunbeauty/data/repository/DataStoreRepository.kt (hunk 3 above)
Comment (MaxAstin) [reply data/src/main/java/com/bunbeauty/data/repository/DataStoreRepository.kt:55]: true

**Review by xidealo — APPROVED** [review]

**Pull request merged**